crates/bevy_ecs/src/schedule/executor.rs

```diff
@@ -10,25 +10,15 @@ pub trait ParallelSystemExecutor: Downcast + Send + Sync {
 
 impl_downcast!(ParallelSystemExecutor);
 
-pub struct SingleThreadedExecutor {
-    /// Last archetypes generation observed by parallel systems.
-    archetype_generation: ArchetypeGeneration,
-}
+#[derive(Default)]
+pub struct SingleThreadedExecutor;
 
-impl Default for SingleThreadedExecutor {
-    fn default() -> Self {
-        Self {
-            archetype_generation: ArchetypeGeneration::initial(),
-        }
-    }
-}
 impl ParallelSystemExecutor for SingleThreadedExecutor {
     fn rebuild_cached_data(&mut self, _: &[ParallelSystemContainer]) {}
 
     fn run_systems(&mut self, systems: &mut [ParallelSystemContainer], world: &mut World) {
-        self.update_archetypes(systems, world);
-
         for system in systems {
+            system.system_mut().update_archetypes(world);
```

Comment:
This wouldn't be needed here since the safe `run` function would do it internally.